### PR TITLE
Pro 4299 pnpm documentation

### PR DESCRIPTION
**Pull request by BoDonkey**
## Summary

*Summarize the changes briefly, including which issue/ticket this resolves. If it closes an existing Github issue, include "Closes #[issue number]"*
This PR adds documentation for using `pnpm`. It is a first pass and will likely need extensive revisions and additional info added. Closes PRO-4299.

## What are the specific steps to test this change?

*For example:*
> 1. Run the website and log in as an admin
> 2. Open a piece manager modal and select several pieces
> 3. Click the "Archive" button on the top left of the manager and confirm that it should proceed
> 4. Check that all pieces have been archived properly
1. Build the documentation
2. Navigate to `/guide/using-pnpm.html`

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [X] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: docs/guide/using-pnpm.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+:::
+
+When creating a project from scratch, Apostrophe will attempt to automatically detect the use of `pnpm` by examining your project root directory for a `pnpm-lock.yaml` file. Some monorepo tools will interfere with automatic detection. You can force the use of `pnpm` by setting an option of `pnpm: 'true'` in your `app.js` file. 
+
+When using the Apostrophe CLI tool to create a new project `npm` will be used by default to install your dependencies. This means that if you create a project using the CLI you will need to convert it to utilize `pnpm`.
+
+This is as simple as deleting the `node_modules` folder and the `package-lock.json` files. Additionally, you want to change the `release` script of your `package.json` file to use `pnpm` instead of `npm`.
+
+Finally to install your packages run `pnpm install`. You should be sure to install with peer dependencies, either by passing the configuration in using `pnpm install --config.auto-install-peers=true`, or by setting it using `pnpm set auto-install-peers=true --global`. Using global will force this configuration on all projects built locally and you can leave this off.
```
Comment (boutell) [inline docs/guide/using-pnpm.md:15]: Is this part necessary in all versions of pnpm? Did Miro say the default is to do it in latest pnpm?

**Review by boutell — CHANGES_REQUESTED** [review]

File: docs/guide/using-pnpm.md (hunk 1 above)
Comment (BoDonkey) [reply docs/guide/using-pnpm.md:15]: I found that I didn't need it in v8, but Miro suggested that it wasn't bad to include just in case.

Comment (myovchev) [reply docs/guide/using-pnpm.md:15]: I haven't found any source suggesting "best practices" when it comes to moving from npm to pnpm.

**Review by myovchev — APPROVED** [review]

**Review by boutell — APPROVED** [review]
Approving — no strong feeling about the defaults and v8 thing, note it or do not, I can see it either way

**Review by boutell — APPROVED** [review]

**Pull request merged**